src/webgpu/api/validation/copyBufferToBuffer.spec.ts

```diff
@@ -0,0 +1,263 @@
+    });
+
+    t.TestCopyBufferToBuffer({
+      srcBuffer,
+      srcOffset,
+      dstBuffer,
+      dstOffset,
+      copySize,
+      isSuccess,
+    });
+  });
+
+g.test('copy_size')
+  .params([
+    { copySize: 0, isSuccess: true },
+    { copySize: 2, isSuccess: false },
+    { copySize: 4, isSuccess: true },
+    { copySize: 5, isSuccess: false },
+    { copySize: 8, isSuccess: true },
+  ] as const)
+  .fn(async t => {
+    const { copySize, isSuccess } = t.params;
+
+    const bufferSize = 16;
+    const srcOffset = 0;
+    const dstOffset = 0;
+
+    const srcBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_SRC,
+    });
+    const dstBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_DST,
+    });
+
+    t.TestCopyBufferToBuffer({
+      srcBuffer,
+      srcOffset,
+      dstBuffer,
+      dstOffset,
+      copySize,
+      isSuccess,
+    });
+  });
+
+g.test('copy_offset')
+  .params([
+    { srcOffset: 0, dstOffset: 0, isSuccess: true },
+    { srcOffset: 2, dstOffset: 0, isSuccess: false },
+    { srcOffset: 4, dstOffset: 0, isSuccess: true },
+    { srcOffset: 5, dstOffset: 0, isSuccess: false },
+    { srcOffset: 8, dstOffset: 0, isSuccess: true },
+    { srcOffset: 0, dstOffset: 2, isSuccess: false },
+    { srcOffset: 0, dstOffset: 4, isSuccess: true },
+    { srcOffset: 0, dstOffset: 5, isSuccess: false },
+    { srcOffset: 0, dstOffset: 8, isSuccess: true },
+    { srcOffset: 4, dstOffset: 8, isSuccess: true },
+  ] as const)
+  .fn(async t => {
+    const { srcOffset, dstOffset, isSuccess } = t.params;
+
+    const bufferSize = 16;
+    const copySize = 8;
+
+    const srcBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_SRC,
+    });
+    const dstBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_DST,
+    });
+
+    t.TestCopyBufferToBuffer({
+      srcBuffer,
+      srcOffset,
+      dstBuffer,
+      dstOffset,
+      copySize,
+      isSuccess,
+    });
+  });
+
+g.test('copy_overflow')
+  .params([
+    { srcOffset: 16, dstOffset: 0, copySize: MAX_ALIGNED_SAFE_INTEGER },
+    { srcOffset: 0, dstOffset: 16, copySize: MAX_ALIGNED_SAFE_INTEGER },
+    { srcOffset: MAX_ALIGNED_SAFE_INTEGER, dstOffset: 0, copySize: 16 },
+    { srcOffset: 0, dstOffset: MAX_ALIGNED_SAFE_INTEGER, copySize: 16 },
+  ] as const)
+  .fn(async t => {
+    const { srcOffset, dstOffset, copySize } = t.params;
+
+    const bufferSize = 16;
+    const srcBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_SRC,
+    });
+    const dstBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_DST,
+    });
+
+    t.TestCopyBufferToBuffer({
+      srcBuffer,
+      srcOffset,
+      dstBuffer,
+      dstOffset,
+      copySize,
+      isSuccess: false,
+    });
+  });
+
+g.test('copy_out_of_bounds')
+  .params([
```

Comment:
Would be best to have a base case which does succeed, to be more sure the rest of the test cases are testing what they intend to test.
```suggestion
  .params([
    { srcOffset: 0, dstOffset: 0, copySize: 32, _success: true },
```

Reply:
Done